### PR TITLE
feat: Expose Config RouteAction.IdleTimeout

### PR DESCRIPTION
**Pull request by hunshcn**
Signed-off-by: hunshcn <hunsh.cn@gmail.com>

# Description

Adds the new IdleTimeout field to RouteOptions to  correspond to envoys config.route.v3.RouteAction.IdleTimeout

# Context

> [gloo http_connection_manager streamIdleTimeout](https://docs.solo.io/gloo-edge/latest/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/hcm/hcm.proto.sk/#httpconnectionmanagersettings) so as to [envoy stream_idle_timeout](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-stream-idle-timeout) can define the timeout of long connection without upstream and downstream traffic.

https://github.com/solo-io/gloo/issues/7545

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/7545

### CODE REVIEW

File: projects/gloo/api/v1/options.proto [hunk 1]
```diff
@@ -475,6 +475,7 @@ message RouteOptions {
     // Settings for maximum durations and timeouts for streams on the route.
     // Please refer to the [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-routeaction-maxstreamduration)
     MaxStreamDuration max_stream_duration = 28;
+    google.protobuf.Duration idle_timeout = 29;
```
Comment (jackstine) [inline projects/gloo/api/v1/options.proto:478]: I would like to expose some comments about this attribute. based off the Envoy documentation, as well as a link to the docs.

https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-stream-idle-timeout

Comment (hunshcn) [reply projects/gloo/api/v1/options.proto:482]: > I would like to expose some comments about this attribute. based off the Envoy documentation, as well as a link to the docs.
> 
> https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-stream-idle-timeout

thx, comment now

**Comment by jackstine** [comment]
You will have to run `make generate-all -B` or `make generated-code -B` to fix the code gen.  Some doc files will need to be updated. Once that is done, we can approve this PR.

**Comment by hunshcn** [comment]
codegen done @jackstine

**Review by jackstine — APPROVED** [review]
🚀

**Comment by sheidkamp** [comment]
Mirroring a comment on one of my recent PRs... e2e test?

**Review by sheidkamp — APPROVED** [review]
LGTM, left a comment about e2e tests, but those don't seem to be the norm for this type of change atm.

**Pull request merged**